### PR TITLE
Add cnvkit germline mode

### PR DESCRIPTION
**Pull request by SusiJo**
## PR checklist

This PR implements germline variant calling with cnvkit.

- [x] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
    - [ ] If you've added a new tool - add to the software_versions process and a regex to `scrape_software_versions.py`
    - [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/sarek/tree/master/.github/CONTRIBUTING.md)
    - [ ] If necessary, also make a PR on the nf-core/sarek _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [x] Make sure your code lints (`nf-core lint .`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [x] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

**Comment by SusiJo** [comment]
Please note that the new upstream `nf-core/modules` `cnvkit/batch` has to be installed after the [PR](https://github.com/nf-core/modules/pull/1719) to `modules` is merged.

File: subworkflows/local/germline_variant_calling.nf [hunk 1]
```diff
@@ -62,6 +64,24 @@ workflow GERMLINE_VARIANT_CALLING {
             cram, crai, bed_new, tbi_new]
         }
 
+    // CNVKIT
+    cram_recalibrated.view()
+
```
Comment (maxulysse) [inline subworkflows/local/germline_variant_calling.nf:69]:
```suggestion
```

File: subworkflows/local/germline_variant_calling.nf [hunk 2]
```diff
@@ -62,6 +64,24 @@ workflow GERMLINE_VARIANT_CALLING {
...
+    if(params.tools.contains('cnvkit')){
+        cram_recalibrated_cnvkit = cram_recalibrated
+            .map{ meta, cram, crai ->
+                [meta, [], cram]
+            }
+
+        cram_recalibrated_cnvkit.view()
+
```
Comment (maxulysse) [inline subworkflows/local/germline_variant_calling.nf:77]:
```suggestion
```

**Review by FriederikeHanssen — COMMENTED** [review]
Looks great, only thing missing would be some CI tests. 🚀

**Comment by maxulysse** [comment]
@SusiJo you can run prettier with `@nf-core-bot fix linting`

**Comment by SusiJo** [comment]
Ah cool! Good to know for next time :D

File: subworkflows/local/germline_variant_calling.nf [hunk 3]
```diff
@@ -21,11 +22,14 @@ workflow GERMLINE_VARIANT_CALLING {
...
         intervals_bed_combine_gz_tbi // channel: [mandatory] intervals/target regions index zipped and indexed in one file
         intervals_bed_combine_gz     // channel: [mandatory] intervals/target regions index zipped in one file
+        intervals_bed_combined       // channel: [mandatory] intervals/target regions in one file unzipped
         // joint_germline               // val: true/false on whether to run joint_germline calling, only works in combination with haplotypecaller at the moment
 
     main:
 
     ch_versions          = Channel.empty()
+    print "cram recalibrated original germline"
+    cram_recalibrated.view()
```
Comment (maxulysse) [inline subworkflows/local/germline_variant_calling.nf:32]:
```suggestion
```

Comment (SusiJo) [reply subworkflows/local/germline_variant_calling.nf:32]: weird, i removed that before..

File: tests/test_tools.yml [hunk 4]
```diff
@@ -1,3 +1,75 @@
...
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted.targetcoverage.cnn
+    - path: results/variant_calling/sample1/cnvkit/reference.cnn
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted-diagram.pdf
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted-scatter.png
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted.bintest.cns
+    - path: results/variant_calling/sample1/cnvkit/multi_intervals.target.bed
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted.cnr
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted.cns
+    - path: results/variant_calling/sample1/cnvkit/test.paired_end.recalibrated.sorted.call.cns
+
```
Comment (FriederikeHanssen) [inline tests/test_tools.yml:72]: looks great, thanks for adding those <3

**Comment by maxulysse** [comment]
CI tests are badly failing

**Comment by FriederikeHanssen** [comment]
linting fails for cnvkit bathc:

```
│ cnvkit/batch                              │ modules/nf-co… │ Local copy of   │
│                                           │                │ module does not │
│                                           │                │ match remote    
```
could that be the reason?

**Comment by maxulysse** [comment]
Seems likely, anyway error message is not helping as usual...

**Review by maxulysse — APPROVED** [review]

**Pull request merged**